### PR TITLE
Ruby Ruby Ruby Ruby!

### PR DESCRIPTION
**Pull request by henryaj**
🎶 Let it never be said that romance is dead 
'Cause there's so little else occupying my head 🎶

### CODE REVIEW

**Comment by dan-f** [comment]
Whoops, looks like I didn't get a notification. Wonder how I can get GH to email me when people open PRs...

Mind doing a refactor or merge to resolve the conflict?

**Comment by henryaj** [comment]
Done. Should be good to merge.

File: wc-ruby/lib/wc-ruby.rb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  }
+
+  threads.map(&:join)
+
+  Process.waitall
+
+  return results
+end
+
+def create_socket_slave(basepath)
```
Comment (dan-f) [inline wc-ruby/lib/wc-ruby.rb]: Let's use a different name for this method. Worker?

File: wc-ruby/lib/wc-ruby.rb [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      end
+    end
+  end
+
+  threads.map(&:join)
+
+  return results
+end
+
+# using processes (system threads)
```
[the remark this answers is not in the document]
Comment (henryaj) [reply wc-ruby/lib/wc-ruby.rb]: Thanks for the clarification. In this case these are distinct OS processes, not threads, so will update accordingly.

**Comment by henryaj** [comment]
Done

**Pull request merged**